lesson23/script.js

```diff
@@ -0,0 +1,140 @@
+// 1.Запросить у пользователя его возраст и определить, кем он является: ребенком (0–2), подростком (12–18), взрослым (18_60) или пенсионером (60– ...).
+
+const age = +prompt("Сколько тебе лет?");
+if (age >= 0 && age < 12) {
+  alert(`Ты ребенок`);
+} else if (age >= 12 && age < 18) {
+  alert(`Ты тинейджер`);
+} else if (age >= 18 && age < 60) {
+  alert(`Ты взрослый`);
+} else if (age >= 60 && age < 150) {
+  alert(`Ты старик`);
+} else if (age >= 150) {
+  alert(`Поздравляем вы Джедай!`);
+}
+
+// 2.Запросить у пользователя число от 0 до 9 и вывести ему спецсимвол, который расположен на этой клавише (1–!, 2–@, 3–# и т. д).
+
+const number = +prompt("Введите число от 0 до 9");
+switch (number) {
+  case "0":
+    alert(`0 is :)`);
+    break;
+  case "1":
+    alert(`1 is !`);
+    break;
+  case "2":
+    alert(`2 is @`);
+    break;
+  case "3":
+    alert(`3 is #`);
+    break;
+  case "4":
+    alert(`4 is $`);
+    break;
+  case "5":
+    alert(`5 is %`);
+    break;
+  case "6":
+    alert(`6 is ^`);
+    break;
+  case "7":
+    alert(`7 is &`);
+    break;
+  case "8":
+    alert(`8 is *`);
+    break;
+  case "9":
+    alert(`9 is (`);
+    break;
+  default:
+    alert(`Введите корректное число`);
+}
+
+// 3.Запросить у пользователя трехзначное число и проверить, есть ли в нем одинаковые цифры.
+
+const number = +prompt("Введите трехзначное число");
+if (
+  number[0] == number[1] ||
+  number[0] == number[2] ||
+  number[1] == number[2]
+) {
+  alert(`TRUE (Здесь есть одинаковые цифры)`);
+} else alert(`FALSE (Здесь нету одинаковых цифр)`);
+
+// 4.Запросить у пользователя год и проверить, високосный он или нет. Високосный год либо кратен 400, либо кратен 4 и при этом не кратен 100.
+
+const year = +prompt("Введите Год:");
+if (year % 4 == 0 && year % 100 !== 0) {

```

Comment:
нужно ещё учесть кратность 400
```suggestion
if (year % 400 || (year % 4 == 0 && year % 100 !== 0)) {

```